src/offcanvas/offcanvas-stack.ts

```diff
@@ -0,0 +1,201 @@
+    const contentRef =
+        this._getContentRef(moduleCFR, options.injector || contentInjector, content, activeOffcanvas, options);
+
+    let backdropCmptRef: ComponentRef<NgbOffcanvasBackdrop>| undefined =
+        options.backdrop !== false ? this._attachBackdrop(moduleCFR, containerEl) : undefined;
+    let panelCmptRef: ComponentRef<NgbOffcanvasPanel> = this._attachWindowComponent(moduleCFR, containerEl, contentRef);
+    let ngbOffcanvasRef: NgbOffcanvasRef =
+        new NgbOffcanvasRef(panelCmptRef, contentRef, backdropCmptRef, options.beforeDismiss);
+
+    this._registerOffcanvasRef(ngbOffcanvasRef);
+    this._registerPanelCmpt(panelCmptRef);
+    ngbOffcanvasRef.hidden.pipe(finalize(() => this._restoreScrollBar())).subscribe();
+    activeOffcanvas.close = (result: any) => { ngbOffcanvasRef.close(result); };
+    activeOffcanvas.dismiss = (reason: any) => { ngbOffcanvasRef.dismiss(reason); };
+
+    this._applyPanelOptions(panelCmptRef.instance, options);
+
+    if (backdropCmptRef && backdropCmptRef.instance) {
+      this._applyBackdropOptions(backdropCmptRef.instance, options);
+      backdropCmptRef.changeDetectorRef.detectChanges();
+    }
+    panelCmptRef.changeDetectorRef.detectChanges();
+    return ngbOffcanvasRef;
+  }
+
+  get activeInstance() { return this._activeInstance; }
+
+  dismiss(reason?: any) { this._offcanvasRef ?.dismiss(reason); }
+
+  hasOpenOffcanvas(): boolean { return !!this._offcanvasRef; }
+
+  private _attachBackdrop(moduleCFR: ComponentFactoryResolver, containerEl: any): ComponentRef<NgbOffcanvasBackdrop> {
+    let backdropFactory = moduleCFR.resolveComponentFactory(NgbOffcanvasBackdrop);
+    let backdropCmptRef = backdropFactory.create(this._injector);
+    this._applicationRef.attachView(backdropCmptRef.hostView);
+    containerEl.appendChild(backdropCmptRef.location.nativeElement);
+    return backdropCmptRef;
+  }
+
+  private _attachWindowComponent(moduleCFR: ComponentFactoryResolver, containerEl: any, contentRef: any):
+      ComponentRef<NgbOffcanvasPanel> {
+    let panelFactory = moduleCFR.resolveComponentFactory(NgbOffcanvasPanel);
+    let panelCmptRef = panelFactory.create(this._injector, contentRef.nodes);
+    this._applicationRef.attachView(panelCmptRef.hostView);
+    containerEl.appendChild(panelCmptRef.location.nativeElement);
+    return panelCmptRef;
+  }
+
+  private _applyPanelOptions(windowInstance: NgbOffcanvasPanel, options: NgbOffcanvasOptions): void {
+    this._panelAttributes.forEach((optionName: string) => {
+      if (isDefined(options[optionName])) {
+        windowInstance[optionName] = options[optionName];
+      }
+    });
+  }
+
+  private _applyBackdropOptions(backdropInstance: NgbOffcanvasBackdrop, options: NgbOffcanvasOptions): void {
+    this._backdropAttributes.forEach((optionName: string) => {
+      if (isDefined(options[optionName])) {
+        backdropInstance[optionName] = options[optionName];
+      }
+    });
```

Comment:
```suggestion
    this._backdropAttributes.forEach(optionName => backdropInstance[optionName] ??= options[optionName]);
```
this should work exactly same way, skipping `isDefined` details

Reply:
disregard, I misread the purpose